### PR TITLE
Video listing page layout

### PR DESCRIPTION
**Pull request by zoef-yt**
In this pr, I added explore page with the category selection bar. The app is mobile-first so you can check it on any screen. There is a little animation on the hover of the video card and the card has a shadow in light theme.

[Files to check](https://github.com/zoef-yt/game-plex-react/pull/2/commits/de5dcbec51d7e6b198b0c77444db04fe8bcafffc?file-filters%5B%5D=.css&file-filters%5B%5D=.jsx&show-viewed-files=true)

[Preview](https://deploy-preview-2--game-plex-zoef.netlify.app/)

### CODE REVIEW

File: src/Pages/VideoListingPage/VideoListingPage.css [hunk 1]
```diff
@@ -0,0 +1,131 @@
+.video-listing-section {
+	display: flex;
+	flex-wrap: wrap;
+	justify-content: flex-start;
+	margin: 0 auto;
+	max-width: 1560px;
```
Comment (zoyatanweer) [inline src/Pages/VideoListingPage/VideoListingPage.css:6]: use rem

File: src/Pages/VideoListingPage/VideoListingPage.css [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+	overflow: hidden;
+	margin: 0 2rem 2rem 0;
+	min-width: 24rem;
+	max-width: 29rem;
+	background-color: var(--colour-black-fifth);
+	border-radius: 0.5rem;
+	cursor: pointer;
+	transition: all 0.3s ease-in-out;
+	box-shadow: 0 2px 5px var(--colour-dark-with-opacity);
+	perspective: 400px;
```
Comment (srishtigautam21) [inline src/Pages/VideoListingPage/VideoListingPage.css:20]: use rem

**Review by srishtigautam21 — APPROVED** [review]
clean code
good to merge with changes

**Review by zoyatanweer — APPROVED** [review]
- code looks great and is easy to understand

- few suggestion in UI : 
      1. you can make the first video which is on display in the homepage less wider, maybe make it the same width as the featured categories(from start to end). This will make the page look more uniform.
      2. When hovering on the featured categories, the cursor is in pointer when hovered on the "watch now" section. You can make the cursor as pointer on the whole video so the user doesn't has to specifically click on "watch now".
      
Appreciation: Love the animation . Great UI .

-good to merge

File: src/Context/CategoryContext/CategoryContext.jsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { createContext, useState, useEffect, useContext } from 'react';
+import { useAxios } from '../../CustomHooks/useAxios';
+
+const CategoryContext = createContext();
```
Comment (muskanbagrecha) [inline src/Context/CategoryContext/CategoryContext.jsx:4]: You can give an empty array as the initial value.

File: src/Context/CategoryContext/CategoryContext.jsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+const CategoryProvider = ({ children }) => {
+	const [categories, setCategories] = useState('');
```
Comment (muskanbagrecha) [inline src/Context/CategoryContext/CategoryContext.jsx:7]: You can use an empty array as the initial value.

File: src/Context/CategoryContext/CategoryContext.jsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+	const { response: categoriesResponse, operation: fetchCategory, loading: loadingCategories } = useAxios();
+
+	useEffect(() => {
+		if (categoriesResponse?.categories) {
```
Comment (muskanbagrecha) [inline src/Context/CategoryContext/CategoryContext.jsx:12]: Can you ternary.

File: src/Context/VideoContext/VideoContext.jsx [hunk 6]
```diff
@@ -0,0 +1,29 @@
+import { createContext, useState, useEffect, useContext } from 'react';
+import { useAxios } from '../../CustomHooks/useAxios';
+
+const VideoContext = createContext();
+
+const VideoProvider = ({ children }) => {
+	const [videos, setVideos] = useState('');
```
Comment (muskanbagrecha) [inline src/Context/VideoContext/VideoContext.jsx:7]: Empty array as the initial value.

File: src/Context/VideoContext/VideoContext.jsx [hunk 7]
```diff
@@ -0,0 +1,29 @@
+import { createContext, useState, useEffect, useContext } from 'react';
+import { useAxios } from '../../CustomHooks/useAxios';
+
+const VideoContext = createContext();
```
Comment (muskanbagrecha) [inline src/Context/VideoContext/VideoContext.jsx:4]: Empty array as the initial value.

File: src/Context/VideoContext/VideoContext.jsx [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+const VideoProvider = ({ children }) => {
+	const [videos, setVideos] = useState('');
+
+	const { response: videosResponse, operation: fetchVideo, loading: loadingVideos } = useAxios();
+
+	useEffect(() => {
+		if (videosResponse?.videos) {
```
Comment (muskanbagrecha) [inline src/Context/VideoContext/VideoContext.jsx:12]: ternary

File: src/Pages/PageNotFound/PageNotFound.jsx [hunk 9]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { useNavigate } from 'react-router-dom';
+import './PageNotFound.css';
+
+function PageNotFound() {
```
Comment (muskanbagrecha) [inline src/Pages/PageNotFound/PageNotFound.jsx:5]: Arrow function.

File: src/Pages/VideoListingPage/VideoListingPage.css [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+}
+
+.video-card {
+	position: relative;
+	overflow: hidden;
+	margin: 0 2rem 2rem 0;
+	min-width: 24rem;
+	max-width: 29rem;
+	background-color: var(--colour-black-fifth);
+	border-radius: 0.5rem;
```
Comment (muskanbagrecha) [inline src/Pages/VideoListingPage/VideoListingPage.css:16]: You can use px in border radius

File: src/Pages/VideoListingPage/VideoListingPage.css [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+	display: flex;
+	justify-content: center;
+	align-items: center;
+	top: 0;
+	right: 0;
+	padding: 5px;
+	background-color: var(--colour-accent-third);
+	color: var(--both-theme-white);
+	transform-origin: right;
+	transform: rotateY(-120deg);
```
Comment (muskanbagrecha) [inline src/Pages/VideoListingPage/VideoListingPage.css:38]: Appreciation: great hover effect.

File: src/Pages/VideoListingPage/VideoListingPage.css (hunk 2 above)
Comment (muskanbagrecha) [inline src/Pages/VideoListingPage/VideoListingPage.css:20]: Appreciation: Great use of perspective

File: src/Pages/VideoListingPage/components/CategorySelectionBar.css [hunk 12]
```diff
@@ -0,0 +1,30 @@
+.category-section {
+	position: -webkit-sticky;
+	position: sticky;
+	top: 0rem;
```
Comment (muskanbagrecha) [inline src/Pages/VideoListingPage/components/CategorySelectionBar.css:4]: Remove rem.

**Review by muskanbagrecha — CHANGES_REQUESTED** [review]
Great work. Do look into the suggested changes.

File: src/Pages/VideoListingPage/VideoListingPage.css (hunk 2 above)
Comment (zoef-yt) [reply src/Pages/VideoListingPage/VideoListingPage.css:20]: done

File: src/Pages/VideoListingPage/VideoListingPage.css (hunk 1 above)
Comment (zoef-yt) [reply src/Pages/VideoListingPage/VideoListingPage.css:6]: done

File: src/Context/CategoryContext/CategoryContext.jsx (hunk 3 above)
Comment (zoef-yt) [reply src/Context/CategoryContext/CategoryContext.jsx:4]: done.

File: src/Context/CategoryContext/CategoryContext.jsx (hunk 4 above)
Comment (zoef-yt) [reply src/Context/CategoryContext/CategoryContext.jsx:7]: done.

File: src/Context/CategoryContext/CategoryContext.jsx (hunk 5 above)
Comment (zoef-yt) [reply src/Context/CategoryContext/CategoryContext.jsx:12]: done.

File: src/Context/VideoContext/VideoContext.jsx (hunk 6 above)
Comment (zoef-yt) [reply src/Context/VideoContext/VideoContext.jsx:7]: done.

File: src/Context/VideoContext/VideoContext.jsx (hunk 7 above)
Comment (zoef-yt) [reply src/Context/VideoContext/VideoContext.jsx:4]: done.

File: src/Context/VideoContext/VideoContext.jsx (hunk 8 above)
Comment (zoef-yt) [reply src/Context/VideoContext/VideoContext.jsx:12]: done.

File: src/Pages/PageNotFound/PageNotFound.jsx (hunk 9 above)
Comment (zoef-yt) [reply src/Pages/PageNotFound/PageNotFound.jsx:5]: done.

File: src/Pages/VideoListingPage/VideoListingPage.css (hunk 10 above)
Comment (zoef-yt) [reply src/Pages/VideoListingPage/VideoListingPage.css:16]: done.

File: src/Pages/VideoListingPage/VideoListingPage.css (hunk 11 above)
Comment (zoef-yt) [reply src/Pages/VideoListingPage/VideoListingPage.css:38]: Thanks.

File: src/Pages/VideoListingPage/VideoListingPage.css (hunk 2 above)
Comment (zoef-yt) [reply src/Pages/VideoListingPage/VideoListingPage.css:20]: Thanks

File: src/Pages/VideoListingPage/components/CategorySelectionBar.css (hunk 12 above)
Comment (zoef-yt) [reply src/Pages/VideoListingPage/components/CategorySelectionBar.css:4]: done.

**Comment by zoef-yt** [comment]
> * code looks great and is easy to understand
> * few suggestion in UI :
>   1. you can make the first video which is on display in the homepage less wider, maybe make it the same width as the featured categories(from start to end). This will make the page look more uniform.
>   2. When hovering on the featured categories, the cursor is in pointer when hovered on the "watch now" section. You can make the cursor as pointer on the whole video so the user doesn't has to specifically click on "watch now".
> 
> Appreciation: Love the animation . Great UI .
> 
> -good to merge

Thanks for the suggestion will look into it.

**Comment by zoef-yt** [comment]
@muskanbagrecha did all the suggested changes in #3.

**Comment by muskanbagrecha** [comment]
Good to merge as all the changes have been made in the next PR. Overall, love the UI and lots to learn from your animation effects. Simply brilliant. Great work on the layout of the listing page as it looks good on all screens.

**Review by muskanbagrecha — APPROVED** [review]
Good to merge as all the changes have been made in the next PR. Overall, love the UI and lots to learn from your animation effects. Simply brilliant. Great work on the layout of the listing page as it looks good on all screens.

**Pull request merged**